src/app/features/Chanels/redux/actions.js

```diff
@@ -0,0 +1,41 @@
+import { GET_CHANNELS, GET_POSTS } from './actionsTypes';
+import { requestChannels, requestPosts } from '../service';
+
+export const getChannels = (channels) => ({
+  type: GET_CHANNELS,
+  payload: channels,
+});
+
+export const getPosts = (posts) => ({
+  type: GET_POSTS,
+  payload: posts,
+});
+
+export function fetchChannels() {
+  return async (dispatch, getState) => {
+    const channelsExists = getState().channelsState.channels.length > 0;
```

Comment:
could be just 
`getState().channels.length `

Try to keep state as flat as possible. Avoid redundant nesting.